hicexplorer/chicPlotViewpoint.py

```diff
@@ -0,0 +1,179 @@
+                        line_data_raw.append(background_data[key][0])
+                        interaction_file_data_raw[key] = line_data_raw
+
+                        line_data_raw = interaction_file_data_raw[key]
+                        line_data_raw.append(background_data[key][1])
+                        interaction_file_data_raw[key] = line_data_raw
+
+        else:
+            data = []
+            interaction_key = sorted(interaction_data)
+            for key in interaction_key:
+                data.append(interaction_data[key])
+            log.debug('data {}'.format(interaction_key))
+            viewpoint_index = interaction_key.index(0)
+
+        # legend = [matrix_name, 'background model']
+        data_plot_label = ax.plot(range(len(data)), data, alpha=0.9, label=matrix_name)
+
+        # z_score
+        ax_sub = ax.twinx()
+        data_plot_label += ax_sub.plot(range(len(z_score)), z_score, '--y', alpha=0.7, label='z-score')
+        ax_sub.set_ylabel('z-score', color='y')
+
+        if args.backgroundModelFile:
+            ax_sub_background = ax.twinx()
+            ax_sub_background.spines["right"].set_position(("axes", 1.1))
+            ax_sub_background.set_ylabel('background model', color='r')
+
+            data_background = np.array(data_background)
+            data_background_mean = np.array(data_background_mean)
+            data_plot_label += ax_sub_background.plot(range(len(data_background)), data_background, '--r', alpha=0.5, label='background model')
+            ax_sub_background.fill_between(range(len(data_background)), data_background + data_background_mean, data_background - data_background_mean, facecolor='red', alpha=0.3)
+
+        ax.set_xticks([0, viewpoint_index, len(data)])
+
+        ax.set_xticklabels([upstream_range, referencePointString, downstream_range])
+        ax.set_ylabel('Number of interactions')
+
+        # multiple legends in one figure
+        data_legend = [label.get_label() for label in data_plot_label]
+        ax.legend(data_plot_label, data_legend, loc=0)
+
+        outFileName = '.'.join(args.outFileName.split('.')[:-1])
+        fileFormat = args.outFileName.split('.')[-1]
+
+        plt.savefig(outFileName + '_' + header + '.' + fileFormat, dpi=args.dpi)
+        plt.close(fig)
+
+        if args.outputViewpointFile:
+
+            interaction_file_data_raw_sorted = sorted(interaction_file_data_raw)
+            with open(args.outputViewpointFile, 'w') as output_file:
+                output_file.write('#ChrViewpoint\tStart\tEnd\tChrInteraction\tStart\tEnd\tRelative position\tRelative Interactions\tZ-score')
+
+                if args.backgroundModelFile:
+                    output_file.write('\tbackground model\tbackground model SEM\n')
+                else:
+                    output_file.write('\n')
+                for key in interaction_file_data_raw_sorted:
+
+                    array_ = interaction_file_data_raw[key]
```

Comment:
you don't need the `_`, thats not best practice, if you want to indicate local vars or functions use prefixes ...